src/plugins/kibana/public/dashboard/components/panel/panel.html

```diff
@@ -1,14 +1,34 @@
 <div class="panel panel-default" ng-switch on="panel.type" ng-if="savedObj || error">
   <div class="panel-heading">
-    <span class="panel-title" title="{{::savedObj.title}}">
+    <div class="panel-title" title="{{::savedObj.title}}">
       <i
         class="fa"
         ng-class="savedObj.vis.type.icon"
         aria-label="{{::savedObj.vis.type.title}} Icon"
         title="{{::savedObj.vis.type.title}}">
       </i>
-      {{::savedObj.title}}
-    </span>
+      <a class="action panel-title-edit" ng-hide="showFormTitle" ng-click="setTitle()">
+        <span ng-if="panel.title">{{panel.title}}</span>
+        <span ng-hide="panel.title">{{::savedObj.title}}</span>
+      </a>
+
+      <div>
+       <form class="inline-form" ng-if="showFormTitle" ng-show="chrome.getVisible()" role="form">
+          <input type="text" name="title" ng-model="panel.title" placeholder="{{::savedObj.title}}" input-focus>
+          <a class="title-confirm" ng-click="confirmTitle()">
+            <i class="fa fa-save"></i>
+          </a>
```

Comment:
These buttons need tooltips. See the tooltip directive as used on the buttons next to the query bar